### ISSUE TITLE
Run recipes to test ESMValCore 2.1.0

### ISSUE THREAD

**Issue by valeriupredoi** [ESMValGroup/ESMValCore#812]
Hey guys @ESMValGroup/esmvaltool-developmentteam we are just about to release ESMValCore 2.1.0 on Monday next week (12 Oct 2020) and we need to make sure that all the goodies and improvements we include in the new release don't break anything in the recipes' run, could we please ask you to pick a couple of your favorite recipes and run them and let us know if there's any issue with them, by opening a GitHub issue? Please use the release branch [`release_2.1`](https://github.com/ESMValGroup/ESMValCore/tree/release_2.1) - after you've run this could you pls post here which recipes you run so that other don't run them again :beer: 

The reason why we ask you to do this now, and we have not in the past, is we had @mattiarighi as our testing machine, but sadly he got commandeered by other people and he's got no time to do it anymore :grin: 

For installation, to be able to use the release branch, pls see the [instructions](https://docs.esmvaltool.org/en/latest/community/introduction.html#using-the-development-version-of-the-esmvaltool-core-package) to use the installation from source for ESMValCore. @jvegasbsc rightly points out in his [comment](https://github.com/ESMValGroup/ESMValCore/issues/812#issuecomment-704148505) that the `esmvalcore` version condition needs to change in the ESMValTool environment file as per https://github.com/ESMValGroup/ESMValTool/pull/1839 - note, however, that if you *only* reinstall ESMValCore from source to use the pre-release branch and dont need to reinstall ESMValTool, that version change is not necessary.

- [ ] ├── cmorizers
- [ ] │   └── recipe_era5.yml
- [ ] ├── examples
- [ ] │   ├── recipe_check_obs.yml (@schlunma: Fails with missing data on DKRZ)
- [x] │   ├── recipe_concatenate_exps.yml (@schlunma)
- [x] │   ├── recipe_correlation.yml (@schlunma)
- [ ] │   ├── recipe_extract_shape.yml (@schlunma: Fails because of missing shapefile)
- [ ] │   ├── recipe_julia.yml
- [x] │   ├── recipe_my_personal_diagnostic.yml (@schlunma)
- [x] │   ├── recipe_ncl.yml (@schlunma)
- [x] │   ├── recipe_preprocessor_derive_test.yml (@schlunma)
- [x] │   ├── recipe_preprocessor_test.yml (@schlunma)
- [x] │   ├── recipe_python_object_oriented.yml (@schlunma)
- [x] │   ├── recipe_python.yml (@schlunma)
- [ ] │   ├── recipe_r.yml
- [x] │   └── recipe_variable_groups.yml (@schlunma)
- [ ] ├── hydrology
- [ ] │   ├── recipe_hype.yml
- [ ] │   ├── recipe_lisflood.yml
- [ ] │   ├── recipe_marrmot.yml
- [x] │   ├── recipe_pcrglobwb.yml
- [ ] │   └── recipe_wflow.yml
- [x] ├── recipe_albedolandcover.yml
- [ ] ├── recipe_anav13jclim.yml
- [ ] ├── recipe_arctic_ocean.yml
- [x] ├── recipe_autoassess_radiation_rms_Amon_all.yml
- [x] ├── recipe_autoassess_radiation_rms_Amon_obs.yml
- [x] ├── recipe_autoassess_radiation_rms_cfMon_all.yml
- [x] ├── recipe_autoassess_stratosphere.yml
- [ ] ├── recipe_capacity_factor.yml
- [x] ├── recipe_carvalhais14nat.yml
- [ ] ├── recipe_climwip.yml
- [ ] ├── recipe_clouds_bias.yml
- [ ] ├── recipe_clouds_ipcc.yml
- [ ] ├── recipe_collins13ipcc.yml
- [ ] ├── recipe_combined_indices.yml
- [x] ├── recipe_consecdrydays.yml
- [x] ├── recipe_cox18nature.yml
- [x] ├── recipe_cvdp.yml
- [x] ├── recipe_deangelis15nat.yml
- [ ] ├── recipe_diurnal_temperature_index.yml
- [x] ├── recipe_ecs_constraints.yml
- [ ] ├── recipe_ecs_scatter.yml
- [x] ├── recipe_ecs.yml
- [x] ├── recipe_ensclus.yml
- [ ] ├── recipe_extreme_events.yml
- [ ] ├── recipe_extreme_index.yml
- [x] ├── recipe_eyring06jgr.yml
- [ ] ├── recipe_flato13ipcc.yml
- [ ] ├── recipe_heatwaves_coldwaves.yml
- [ ] ├── recipe_hyint_extreme_events.yml
- [ ] ├── recipe_hyint.yml
- [ ] ├── recipe_kcs.yml
- [ ] ├── recipe_landcover.yml
- [ ] ├── recipe_lauer13jclim.yml
- [x] ├── recipe_li17natcc.yml
- [x] ├── recipe_martin18grl.yml
- [ ] ├── recipe_miles_block.yml
- [ ] ├── recipe_miles_eof.yml
- [ ] ├── recipe_miles_regimes.yml
- [ ] ├── recipe_modes_of_variability.yml
- [ ] ├── recipe_multimodel_products.yml
- [x] ├── recipe_ocean_amoc.yml
- [x] ├── recipe_ocean_bgc.yml
- [x] ├── recipe_ocean_example.yml
- [x] ├── recipe_ocean_ice_extent.yml
- [x] ├── recipe_ocean_Landschuetzer2016.yml
- [x] ├── recipe_ocean_multimap.yml
- [x] ├── recipe_ocean_quadmap.yml
- [x] ├── recipe_ocean_scalar_fields.yml
- [ ] ├── recipe_perfmetrics_CMIP5_4cds.yml
- [ ] ├── recipe_perfmetrics_CMIP5.yml
- [ ] ├── recipe_perfmetrics_land_CMIP5.yml
- [ ] ├── recipe_quantilebias.yml
- [ ] ├── recipe_rainfarm.yml
- [ ] ├── recipe_runoff_et.yml
- [ ] ├── recipe_russell18jgr.yml
- [ ] ├── recipe_seaice_drift.yml
- [x] ├── recipe_seaice_feedback.yml
- [ ] ├── recipe_seaice.yml
- [ ] ├── recipe_shapeselect.yml
- [ ] ├── recipe_smpi_4cds.yml
- [ ] ├── recipe_smpi.yml
- [ ] ├── recipe_snowalbedo.yml
- [x] ├── recipe_spei.yml
- [x] ├── recipe_tcr.yml
- [ ] ├── recipe_thermodyn_diagtool.yml
- [ ] ├── recipe_toymodel.yml
- [x] ├── recipe_validation_CMIP6.yml
- [x] ├── recipe_validation.yml
- [ ] ├── recipe_wenzel14jgr.yml
- [ ] ├── recipe_wenzel16jclim.yml
- [x] ├── recipe_wenzel16nat.yml
- [ ] ├── recipe_williams09climdyn_CREM.yml
- [x] └── recipe_zmnam.yml

Note that it is not necessary to test every single recipe for the ESMValCore release, we've never done this in the past either and automatic tests are reasonably good, out of 10 or so releases it only happened once that we released an ESMValCore version that was broken (this happened because we changed things at the last minute). These tests are just to get some extra confidence that we didn't introduce any bugs.

**Comment by jvegasbsc** [comment]
There is a problem when installing release: we require <2.1. ESMValGroup/ESMValTool#1839 fixes it.

 recipe_seaice_feedback.yml runs ok

**Comment by valeriupredoi** [comment]
cheers @jvegasbsc - good catch! An thanks for running and testing :beer:

**Comment by valeriupredoi** [comment]
cheers @jvegasbsc - good catch! An thanks for running and testing :beer:

**Comment by jvegasbsc** [comment]
recipe_albedolandcover.yml also runs ok

**Comment by hb326** [comment]
recipe_smpi.yml crashes. See attached log.

[main_log.txt](https://github.com/ESMValGroup/ESMValCore/files/5334713/main_log.txt)

@bettina-gier: did you ever have that problem?

**Comment by valeriupredoi** [comment]
that is a dummy error that masks the actual error from regridding:
```
Traceback (most recent call last):
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 730, in _run_task
    output_files = task.run()
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/_task.py", line 242, in run
    self.output_files = self._run(input_files)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 428, in _run
    product.apply(step, self.debug)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 296, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 241, in preprocess
    result.append(_run_preproc_function(function, item, settings))
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 224, in _run_preproc_function
    return function(items, **kwargs)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_regrid.py", line 335, in regrid
    cube = esmpy_regrid(cube, target_grid, scheme)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_regrid_esmpy.py", line 331, in regrid
    regridder = build_regridder(src_rep, dst_rep, method)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_regrid_esmpy.py", line 220, in build_regridder
    regridder = build_regridder_2d(src_rep, dst_rep,
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_regrid_esmpy.py", line 149, in build_regridder_2d
    dst_field = cube_to_empty_field(dst_rep)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_regrid_esmpy.py", line 130, in cube_to_empty_field
    grid = get_grid(*esmpy_coords, circular=circular)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/esmvalcore/preprocessor/_regrid_esmpy.py", line 88, in get_grid
    grid = ESMF.Grid(np.array(esmpy_lat.shape),
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/ESMF/util/decorators.py", line 64, in new_func
    return func(*args, **kwargs)
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/ESMF/api/grid.py", line 385, in __init__
    self._struct = ESMP_GridCreateNoPeriDim(self.max_index,
  File "/apps/jasmin/community/esmvaltool/miniconda3/envs/esmvaltool/lib/python3.8/site-packages/ESMF/interface/cbindings.py", line 560, in ESMP_GridCreateNoPeriDim
    raise ValueError('ESMC_GridCreateNoPeriDim() failed with rc = '+str(rc)+
ValueError: ESMC_GridCreateNoPeriDim() failed with rc = 545.    Please check the log files (named "*ESMF_LogFile").
```
there is a serious issue regridding MRI-CGM3 - @zklaus any clues?

**Comment by valeriupredoi** [comment]
OK ran it three times for MRI-CGCM3/hfds and it succesfully completed the preprocessing, @bettina-gier @hb326 could you guys pls run it again? The ESMPY error was a fluke, hopefully your issue was too, I see no reason why it should fail :beer:

**Comment by bouweandela** [comment]
From the log, it looks like a problem with some model when loading the data, a coordinate is misinterpreted as a cube, probably because some attribute is missing somewhere in the netcdef file and then it's trying to merge that coordinate and the cube with the variable into a single cube, which fails for obvious reasons.

@hb326 Could you please attach the file main_log_debug.txt instead? Then we might be able to figure out what model it is that is causing the problem.

**Comment by valeriupredoi** [comment]
it's MRI-CGCM3/hfds which is not problematic with data off CEDA ESGF on Jasmin, might be the data off the DKRZ ESGF node

**Comment by hb326** [comment]
Sure. Here it is:

[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/5340989/main_log_debug.txt)

**Comment by valeriupredoi** [comment]
all autoassess recipes (bar the unreviewed ones), all validation, and **all** `_ocean*` recipes run with no problems (@ledm owes me a :beer: for this but that's okay, I think he owes me about a dozen so far :grin: )

**Comment by bouweandela** [comment]
@hb326 Could you also share the recipe that you're running? From the log it looks like it's trying to regrid `hfds` from model `MRI-CGCM3` to the grid of model `GFDL-ESM2G`, while the SMPI recipe in the ESMValTool actually has a target grid `1x1`.

**Comment by hb326** [comment]
I run the recipe as it comes in the master branch: 
[https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/recipes/recipe_smpi.yml](url)

**Comment by bascrezee** [comment]
I thought I also quickly test `recipe_smpi.yml` from the master branch. I am testing with `master` branch of ESMValTool and `release_2.1` from ESMValCore (just checkin). This is the error message I get: 
```
 in memory consumption occur.
2020-10-07 14:13:26,492 UTC [36998] ERROR   Program terminated abnormally, see stack trace below for more information
Traceback (most recent call last):
  File "/home/crezees/ESMValCore/esmvalcore/_main.py", line 430, in run
    fire.Fire(ESMValTool())
  File "/net/exo/landclim/crezees/conda/envs/esmvaltool-pub/lib/python3.8/site-packages/fire/core.py", line 138, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/net/exo/landclim/crezees/conda/envs/esmvaltool-pub/lib/python3.8/site-packages/fire/core.py", line 463, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/net/exo/landclim/crezees/conda/envs/esmvaltool-pub/lib/python3.8/site-packages/fire/core.py", line 672, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/crezees/ESMValCore/esmvalcore/_main.py", line 407, in run
    process_recipe(recipe_file=recipe, config_user=cfg)
  File "/home/crezees/ESMValCore/esmvalcore/_main.py", line 94, in process_recipe
    recipe = read_recipe_file(recipe_file, config_user)
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 71, in read_recipe_file
    return Recipe(raw_recipe,
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 951, in __init__
    self.tasks = self.initialize_tasks() if initialize_tasks else None
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 1299, in initialize_tasks
    task = _get_preprocessor_task(
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 917, in _get_preprocessor_task
    task = _get_single_preprocessor_task(
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 776, in _get_single_preprocessor_task
    products = _get_preprocessor_products(
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 733, in _get_preprocessor_products
    _update_target_grid(
  File "/home/crezees/ESMValCore/esmvalcore/_recipe.py", line 198, in _update_target_grid
    parse_cell_spec(settings['regrid']['target_grid'])
  File "/home/crezees/ESMValCore/esmvalcore/preprocessor/_regrid.py", line 87, in parse_cell_spec
    raise ValueError(emsg.format(spec))
ValueError: Invalid MxN cell specification for grid, got 'reference_dataset'.
```

**Comment by valeriupredoi** [comment]
> I run the recipe as it comes in the master branch:
> [https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/recipes/recipe_smpi.yml](url)

@hb326 @bascrezee you guys must pull the latest ESMValTool master, I have this for hfds preproc:
```yaml
 ppNOLEVirreg:
    regrid:
      target_grid: 1x1
      scheme: linear
```
It works fine for me (well, apart from that ESMPY transitory error)

**Comment by hb326** [comment]
I just pulled the latest version of the master branch, and the recipe is still the same as before... Maybe we have to adjust the recipe?

**Comment by katjaweigel** [comment]
recipe_deangelis15nat.yml and recipe_li17natcc.yml run ok. Thanks @jvegasbsc for the fix, didn't understand how to do it without.

**Comment by schlunma** [comment]
I added a checklist with all recipes of the current master to the initial post by @valeriupredoi so we get a nice overview which recipe hasn't been tested yet. If someone intends to do tests but is not finished he/she could add his/her name to the corresponding recipe so that we avoid duplicate efforts.

**Comment by bouweandela** [comment]
Note that it is not necessary to test every single recipe for the ESMValCore release, we've never done this in the past either and automatic tests are reasonably good, out of 10 or so releases it only happened once that we released an ESMValCore version that was broken (this was because I changed things at the last minute). These tests are just to get some extra confidence that we didn't introduce any bugs.

**Comment by schlunma** [comment]
Ah sorry, my mistake. I thought we tested all recipes for v2.0 and have to do this again for v2.1

**Comment by bouweandela** [comment]
Yes, we will test everything again, but that will be in two weeks time for the release of the tool. But it's great that you added an overview, that's still really useful!

**Comment by bouweandela** [comment]
@hb326 @bascrezee I have opened a new issue for the problem with recipe_smpi.yml

**Comment by katjaweigel** [comment]
recipe_martin18grl.yml and  recipe_spei.yml are running

**Comment by valeriupredoi** [comment]
cheers muchly everyone! and thanks for the graph list @schlunma :beer: Keep up the momentum, we are gaining on the enemy (no enemy per se, but I thought if I sounded like Gen. Patton you might run more recipes :rofl: )

**Comment by schlunma** [comment]
Succesfully tested recipe_ecs.yml, recipe_cox18nature.yml, recipe_tcr.yml and recipe_ecs_constraints.yml.

**Comment by Peter9192** [comment]
Representing the hydrology section with a report on `recipe_pcrglobwb.yml`: :heavy_check_mark:

**Comment by remi-kazeroni** [comment]
Successfully tested recipe_consecdrydays.yml, recipe_ensclus.yml and recipe_zmnam.yml .

**Comment by koir-su** [comment]
recipe_carvalhais14nat is running. 🍻

**Comment by zechlau** [comment]
recipe_wenzel16nat is running too

**Comment by valeriupredoi** [comment]
hey guys, cheers muchly for doing all this work, @bouweandela and myself have now released [version 2.1.0](https://github.com/ESMValGroup/ESMValCore/releases/tag/v2.1.0) - next stop, ESMValTool v2.1.0 when we'll really have to run **all** them recipes :beer: (code freeze next Monday, 19 Oct for ESMValTool, testing that week)
